parser_library/test/processing/ca_instr_test.cpp

```diff
@@ -27,341 +27,247 @@ using namespace hlasm_plugin::parser_library::semantics;
-    diagnostic_op_consumer_container diags;
-
-    auto it = ctx.ids().find("var2");
-
-    ASSERT_TRUE(ctx.get_var_sym(it));
-
-    auto tmp = get_var_sym_value(ctx, it, std::vector<int> {}, {}, diags).access_c();
-    EXPECT_EQ(tmp, "avc");
+    EXPECT_EQ(get_var_value<C_t>(a.hlasm_ctx(), "VAR2"), "avc");
 }
 
 TEST(var_concatenation, concatenated_string_dot)
 {
-    std::string input("&var setc 'avc'   \n&var2 setc '&var.-get'");
+    std::string input = R"(
+&VAR SETC 'avc'   
+&VAR2 SETC '&VAR.-get'
+)";
     analyzer a(input);
     a.analyze();
 
-    auto& ctx = a.hlasm_ctx();
-    diagnostic_op_consumer_container diags;
-
-    auto it = ctx.ids().find("var2");
-
-    ASSERT_TRUE(ctx.get_var_sym(it));
-
-    auto tmp = get_var_sym_value(ctx, it, std::vector<int> {}, {}, diags).access_c();
-    EXPECT_EQ(tmp, "avc-get");
+    EXPECT_EQ(get_var_value<C_t>(a.hlasm_ctx(), "VAR2"), "avc-get");
 }
 
 TEST(var_concatenation, concatenated_string_double_dot)
 {
-    std::string input("&var setc 'avc'   \n&var2 setc '&var..'");
+    std::string input = R"(
+&VAR SETC 'avc'   
+&VAR2 SETC '&VAR..'
+)";
     analyzer a(input);
     a.analyze();
 
-    auto& ctx = a.hlasm_ctx();
-    diagnostic_op_consumer_container diags;
-
-    auto it = ctx.ids().find("var2");
-
-    ASSERT_TRUE(ctx.get_var_sym(it));
-
-    auto tmp = get_var_sym_value(ctx, it, std::vector<int> {}, {}, diags).access_c();
-    EXPECT_EQ(tmp, "avc.");
+    EXPECT_EQ(get_var_value<C_t>(a.hlasm_ctx(), "VAR2"), "avc.");
 }
 
 TEST(AGO, extended)
 {
     std::string input(R"(
  AGO (2).a,.b,.c
-.a anop   
-&var1 setb 0
-.b anop
-&var2 setb 0
-.c anop
-&var3 setb 0
+.A ANOP   
+&VAR1 SETB 0
+.B ANOP
+&VAR2 SETB 0
+.C ANOP
+&VAR3 SETB 0
 )");
     analyzer a(input);
     a.analyze();
 
-    auto& ctx = a.hlasm_ctx();
-
-    auto it1 = ctx.ids().add("var1");
-    auto it2 = ctx.ids().add("var2");
-    auto it3 = ctx.ids().add("var3");
-
-    EXPECT_FALSE(ctx.get_var_sym(it1));
-    EXPECT_TRUE(ctx.get_var_sym(it2));
-    EXPECT_TRUE(ctx.get_var_sym(it3));
+    EXPECT_EQ(get_var_value<B_t>(a.hlasm_ctx(), "VAR1"), std::nullopt);
+    EXPECT_EQ(get_var_value<B_t>(a.hlasm_ctx(), "VAR2"), false);
+    EXPECT_EQ(get_var_value<B_t>(a.hlasm_ctx(), "VAR3"), false);
 }
 
 TEST(AGO, extended_fail)
 {
     std::string input(R"(
  AGO (8).a,.b,.c
-.a anop   
-&var1 setb 0
-.b anop
-&var2 setb 0
-.c anop
-&var3 setb 0
+.A ANOP   
+&VAR1 SETB 0
+.B ANOP
+&VAR2 SETB 0
+.C ANOP
+&VAR3 SETB 0
 )");
     analyzer a(input);
     a.analyze();
 
-    auto& ctx = a.hlasm_ctx();
-
-    auto it1 = ctx.ids().add("var1");
-    auto it2 = ctx.ids().add("var2");
-    auto it3 = ctx.ids().add("var3");
-
-    EXPECT_TRUE(ctx.get_var_sym(it1));
-    EXPECT_TRUE(ctx.get_var_sym(it2));
-    EXPECT_TRUE(ctx.get_var_sym(it3));
+    EXPECT_EQ(get_var_value<B_t>(a.hlasm_ctx(), "VAR1"), false);
+    EXPECT_EQ(get_var_value<B_t>(a.hlasm_ctx(), "VAR2"), false);
+    EXPECT_EQ(get_var_value<B_t>(a.hlasm_ctx(), "VAR3"), false);
 }
```

Comment:
is there a diagnostic produced?

Reply:
Nope. Should it be? It seems fine when run on the mainframe

Reply:
I was just not sure... looks fine.